pipelines/batch/src/main/java/com/google/fhir/analytics/JdbcFetchHapi.java

```diff
@@ -172,62 +218,109 @@ public FetchRowsJdbcIo(
 
       builder =
           new StringBuilder(
-              "select td.tag_code, td.tag_display, td.tag_system, td.tag_type, tag.res_id from"
-                  + " hfj_res_tag tag join hfj_tag_def td on tag.tag_id = td.tag_id join"
-                  + " hfj_resource res on tag.res_id = res.res_id where res.res_type = ? and"
+              "SELECT td.tag_code, td.tag_display, td.tag_system, td.tag_type, tag.res_id FROM"
+                  + " hfj_res_tag tag JOIN hfj_tag_def td ON tag.tag_id = td.tag_id JOIN"
+                  + " hfj_resource res ON tag.res_id = res.res_id WHERE res.res_type = ? AND"
                   + " res.res_id % ? = ? ");
       if (since != null && !since.isEmpty()) {
         builder.append("  and res.res_updated > '").append(since).append("'");
       }
       tagQuery = builder.toString();
       log.info("JDBC query for tags: " + tagQuery);
+
+      builder =
+          new StringBuilder(
```

Comment:
**thought:** Weird use of `StringBuilder` to only manually string concat a bunch of strings and then convert to a string via `toString()`. Would be better to just set the string...

Reply:
Yeah I probably did that just to follow the pattern in case there's any conditional logic.